lgc/builder/ImageBuilder.cpp

```diff
@@ -752,14 +752,19 @@ Value *ImageBuilder::CreateImageSampleConvertYCbCr(Type *resultTy, unsigned dim,
   Value *pResult = nullptr;
   Value *ycbcrSamplerDesc = CreateExtractValue(convertingSamplerDesc, 0);
 
-  // Extract YCbCr meta data
-  const SamplerYCbCrConversionMetaData yCbCrMetaData = {
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(4)))->getZExtValue()),
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(5)))->getZExtValue()),
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(6)))->getZExtValue()),
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(7)))->getZExtValue()),
+  // Helper function to extract YCbCr meta data from ycbcrSamplerDesc
+  auto getYCbCrMetaElement = [this, &ycbcrSamplerDesc](unsigned idx) {
+    return static_cast<unsigned>(
+        cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(idx)))->getZExtValue());
```

Comment:
I don't think you need the getInt64(idx). You can just use idx directly.

Also, maybe it would be better to use the syntax to specify the return type for the lambda (-> unsigned) and avoid the static_cast<unsigned>?

Reply:
Agreed on both points. (For the CreateExtractElement, there's a bit of an annoyance in that the overloads clash when you try to pass in a literal 0... you have to use 0llu then. But for an unsigned variable like here, dropping the getInt64 should be fine.)

Reply:
- Thank you for explaining about CreateExtractElement, the current version drops getInt64.
- And the static_cast has been removed now :)